From 8eb1f90c3e894da7f528dbc1c54960ebcf0f759b Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 2 Dec 2021 11:56:05 +0000
Subject: [PATCH] RHEL: Remove the --in-place option

This disables the virt-v2v --in-place option which we do not
wish to support in RHEL.
---
 docs/virt-v2v.pod | 8 --------
 tests/Makefile.am | 1 -
 v2v/v2v.ml        | 8 --------
 3 files changed, 17 deletions(-)

diff --git a/docs/virt-v2v.pod b/docs/virt-v2v.pod
index e5400ca7..fc7e0d82 100644
--- a/docs/virt-v2v.pod
+++ b/docs/virt-v2v.pod
@@ -24,9 +24,6 @@ virtualize those machines (physical to virtual, or p2v).
 To estimate the disk space needed before conversion, see
 L<virt-v2v-inspector(1)>.
 
-For in-place conversion, there is a separate tool called
-L<virt-v2v-in-place(1)>.
-
 =head2 Input and Output
 
 You normally run virt-v2v with several I<-i*> options controlling the
@@ -39,10 +36,6 @@ The input and output sides of virt-v2v are separate and unrelated.
 Virt-v2v can read from any input and write to any output.  Therefore
 these sides of virt-v2v are documented separately in this manual.
 
-Virt-v2v normally copies from the input to the output, called "copying
-mode".  In this case the source guest is always left unchanged.
-In-place conversions may be done using L<virt-v2v-in-place(1)>.
-
 =head2 Other virt-v2v topics
 
 L<virt-v2v-support(1)> — Supported hypervisors, virtualization
@@ -1622,7 +1615,6 @@ L<https://rwmj.wordpress.com/2015/09/18/importing-kvm-guests-to-ovirt-or-rhev/#c
 
 L<virt-p2v(1)>,
 L<virt-v2v-inspector(1)>,
-L<virt-v2v-in-place(1)>,
 L<virt-customize(1)>,
 L<virt-df(1)>,
 L<virt-filesystems(1)>,
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 5c523204..5190d055 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -77,7 +77,6 @@ TESTS = \
 	test-v2v-floppy.sh \
 	test-v2v-i-disk.sh \
 	test-v2v-i-ova.sh \
-	test-v2v-in-place.sh \
 	test-v2v-block-driver.sh \
 	test-v2v-inspector.sh \
 	test-v2v-mac.sh \
diff --git a/v2v/v2v.ml b/v2v/v2v.ml
index 47d2da79..19f08aae 100644
--- a/v2v/v2v.ml
+++ b/v2v/v2v.ml
@@ -217,12 +217,6 @@ let rec main () =
     warning (f_"the --vmtype option has been removed and now does nothing")
   in
 
-  (* Options that are errors. *)
-  let in_place_error _ =
-    error (f_"The --in-place option has been replaced by the \
-              ‘virt-v2v-in-place’ program")
-  in
-
   let argspec = [
     [ L"bandwidth" ], Getopt.String ("bps", set_string_option_once "--bandwidth" bandwidth),
                                     s_"Set bandwidth to bits per sec";
@@ -246,8 +240,6 @@ let rec main () =
                                     s_"Use password from file to connect to input hypervisor";
     [ M"it" ],       Getopt.String ("transport", set_string_option_once "-it" input_transport),
                                     s_"Input transport";
-    [ L"in-place" ], Getopt.Unit in_place_error,
-                                    s_"Use virt-v2v-in-place instead";
     [ L"mac" ],      Getopt.String ("mac:network|bridge|ip:out", add_mac),
       s_"Map NIC to network or bridge or assign static IP";
     [ S 'n'; L"network" ], Getopt.String ("in:out", add_network),
